deepdive1/README.md

```diff
@@ -0,0 +1,265 @@
+# Deep Dive 1 :ocean: :swimming_woman: - Testing
+Testing your app is important for making sure your app functions as expected. Today we'll specifically be talking about Unit Testing, which is a form of automated testing. As the name implies, unit testing is where you write tests small units of your code. For example, lets say I want to test that the fibonacci function I wrote actually works. In plain Java it might look like this:
+
+```
+public boolean fibTest(){
+    int fib = doFibonacci(6);
+    if (fib != 8) return false;
+    else return true;
+}
+```
+Since the 6th fibonacci number is 8, if `fib` is not 8, then my test has failed. In Android, your unit tests will be a little different. Since some things you want to test will be more complex than checking Fibonacci numbers, you'll probably have to use a testing library/framework specific to Android. Even for `fibTest` you can use a libary/framework (a certain library is commonly used for this). More on that later!
+
+## Why are unit tests important?
+Say you wrote 23 units tests in your code. You then add a new feature, and now suddenly 5 of your unit tests aren't passing anymore. Unit tests are a clear indicator to show how solidly built your code is, and are very helpful for debugging. You can quickly catch errors in your code and lets you verify that the logic is correct. They might seem superfluous, but they're easy to implement and prevent you from backtracking/spending a lot of time trying to figure out where things went wrong.
```

Comment:
Some of these paragraphs sound like a transcript of someone talking. This might be more because of personal style differences. If you wanted to cut down on length, you could change these paragraphs.